### PR TITLE
[NFM] Problem: Missing chainmaind CLI level NFT types params validation and amino JSON support

### LINKED ISSUE #224: Problem: Introduce chainmaind CLI level NFT types params validation
The NFT module is missing validation for the params used.

### LINKED ISSUE #259: Problem: Missing NFT amino JSON support

### PR DESCRIPTION
**Pull request by cdc-Hitesh**
Fix #224
Fix #259

Building on top of @calvinaco  PR #260 
This PR contains enhancement on the validation and nft message type.

### CODE REVIEW

**Review by eddycpc — APPROVED** [review]
LGTM

File: lib/src/transaction/msg/ow.types.ts [hunk 1]
```diff
@@ -116,41 +116,65 @@ export const owMsgBeginRedelgateOptions = owStrictObject().exactShape({
...
+    message: (label: string) => `Expected ${label} to contain only lowercase alphanumeric characters`,
+});
+
+const owNFTId = ow.string
+    .minLength(3)
+    .maxLength(64)
+    .validate(owNFTStartWithAlphabetValidatorFn)
+    .validate(owNFTIsAlphaNumericValidatorFn);
+
+const owNFTDenomNameValidator = (val: string) => ({
```
Comment (calvinaco) [inline lib/src/transaction/msg/ow.types.ts:136]:
```suggestion
const owNFTNameValidator = (val: string) => ({
```

File: lib/src/transaction/msg/ow.types.ts [hunk 2]
```diff
@@ -116,41 +116,65 @@ export const owMsgBeginRedelgateOptions = owStrictObject().exactShape({
...
+    message: (label: string) => `Expected ${label} to be non-empty string`,
+});
+
+const URI_MAX_LENGTH = 256;
+
 export const owMsgIssueDenomOptions = owStrictObject().exactShape({
-    id: ow.string,
-    name: ow.string,
+    id: owNFTId,
+    name: ow.string.validate(owNFTDenomNameValidator),
```
Comment (calvinaco) [inline lib/src/transaction/msg/ow.types.ts:145]:
```suggestion
    name: ow.string.validate(owNFTNameValidator),
```

File: lib/src/transaction/msg/ow.types.ts [hunk 3]
```diff
@@ -116,41 +116,65 @@ export const owMsgBeginRedelgateOptions = owStrictObject().exactShape({
...
 });
 
 export const owMsgMintNFTOptions = owStrictObject().exactShape({
-    id: ow.string,
-    denomId: ow.string,
-    name: ow.string,
-    uri: ow.string,
+    id: owNFTId,
+    denomId: owNFTId,
+    name: ow.string.validate(owNFTDenomNameValidator),
```
Comment (calvinaco) [inline lib/src/transaction/msg/ow.types.ts:153]:
```suggestion
    name: ow.string.validate(owNFTNameValidator),
```

File: lib/src/transaction/msg/ow.types.ts [hunk 4]
```diff
@@ -116,41 +116,65 @@ export const owMsgBeginRedelgateOptions = owStrictObject().exactShape({
...
 });
 
 export const owMsgEditNFTOptions = owStrictObject().exactShape({
-    id: ow.string,
-    denomId: ow.string,
-    name: ow.string,
-    uri: ow.string,
+    id: owNFTId,
+    denomId: owNFTId,
+    name: ow.string.validate(owNFTDenomNameValidator),
```
Comment (calvinaco) [inline lib/src/transaction/msg/ow.types.ts:163]:
```suggestion
    name: ow.string.validate(owNFTNameValidator),
```

**Review by calvinaco — APPROVED** [review]
LGTM

**Review by calvinaco — APPROVED** [review]

File: lib/src/transaction/msg/ow.types.ts (hunk 1 above)
Comment (cdc-Hitesh) [reply lib/src/transaction/msg/ow.types.ts]: @calvinaco  I have kept DenomName deliberately, as I understand this validation is specific to `denom` name only.

Comment (cdc-Hitesh) [reply lib/src/transaction/msg/ow.types.ts]: Or this validation stays for any name value?
Btw I agree with this validation to be done on any name value though

**Pull request merged**